# Exploring Ebay Car Dataset


In [1]:
import pandas as pd
import numpy as np

In [2]:
autos = pd.read_csv('autos.csv', encoding='Latin-1')
autos = autos[0:100000] # original dataset was over 370,000 rows so I sampled 50,000
autos.info()
autos.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 20 columns):
dateCrawled            100000 non-null object
name                   100000 non-null object
seller                 100000 non-null object
offerType              100000 non-null object
price                  100000 non-null int64
abtest                 100000 non-null object
vehicleType            89901 non-null object
yearOfRegistration     100000 non-null int64
gearbox                94609 non-null object
powerPS                100000 non-null int64
model                  94508 non-null object
kilometer              100000 non-null int64
monthOfRegistration    100000 non-null int64
fuelType               90946 non-null object
brand                  100000 non-null object
notRepairedDamage      80644 non-null object
dateCreated            100000 non-null object
nrOfPictures           100000 non-null int64
postalCode             100000 non-null int64
lastSeen               10000

dateCrawled                            name  seller offerType  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat   Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat   Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat   Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat   Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat   Angebot   

   price abtest vehicleType  yearOfRegistration    gearbox  powerPS  model  \
0    480   test         NaN                1993    manuell        0   golf   
1  18300   test       coupe                2011    manuell      190    NaN   
2   9800   test         suv                2004  automatik      163  grand   
3   1500   test  kleinwagen                2001    manuell       75   golf   
4   3600   test  kleinwagen                2008    manuell       69  fabia   

   kilometer  monthOfRegistration fuelType       brand notRepairedDamage  \
0     150000                    0   benzin  volkswagen               NaN   
1     125000                    5   diesel        audi                ja   
2     125000                    8   diesel        jeep               NaN   
3     150000                    6   benzin  volkswagen              nein   
4      90000                    7   diesel       skoda              nein   

           dateCreated  nrOfPictures  postalCode             lastSeen  
0  2016-03-24 00:00:00             0       70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00             0       66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00             0       90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00             0       91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00             0       60437  2016-04-06 10:17:21

# Clean Columns to be snakecase

In [3]:
autos.columns # get names of current columns to see which ones need changed

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'kilometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [4]:
autos.columns = ['date_crawled', 'name', 'seller', 'offer_type', 'price', 'ab_test',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'num_photos', 'postal_code',
       'last_seen']
autos.head()

date_crawled                            name  seller offer_type  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat    Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat    Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat    Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat    Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat    Angebot   

   price ab_test vehicle_type  registration_year    gearbox  power_ps  model  \
0    480    test          NaN               1993    manuell         0   golf   
1  18300    test        coupe               2011    manuell       190    NaN   
2   9800    test          suv               2004  automatik       163  grand   
3   1500    test   kleinwagen               2001    manuell        75   golf   
4   3600    test   kleinwagen               2008    manuell        69  fabia   

   odometer  registration_month fuel_type       brand unrepaired_damage  \
0    150000                   0    benzin  volkswagen               NaN   
1    125000                   5    diesel        audi                ja   
2    125000                   8    diesel        jeep               NaN   
3    150000                   6    benzin  volkswagen              nein   
4     90000                   7    diesel       skoda              nein   

            ad_created  num_photos  postal_code            last_seen  
0  2016-03-24 00:00:00           0        70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00           0        66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00           0        90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00           0        91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00           0        60437  2016-04-06 10:17:21

# Initial Data Exploration
## Examples of Columns that need cleaned:
 - Any Columns with mostly one value are likely to be dropped.
 - Any columns that need more investigation.
 - Any examples of numeric data stored as text that needs to be cleaned.

In [5]:
autos.describe(include= 'all')

date_crawled      name  seller offer_type         price  \
count                100000    100000  100000     100000  1.000000e+05   
unique                92921     72510       2          2           NaN   
top     2016-03-07 15:48:46  BMW_318i  privat    Angebot           NaN   
freq                      4       183   99999      99998           NaN   
mean                    NaN       NaN     NaN        NaN  1.009095e+04   
std                     NaN       NaN     NaN        NaN  5.634880e+05   
min                     NaN       NaN     NaN        NaN  0.000000e+00   
25%                     NaN       NaN     NaN        NaN  1.150000e+03   
50%                     NaN       NaN     NaN        NaN  2.950000e+03   
75%                     NaN       NaN     NaN        NaN  7.200000e+03   
max                     NaN       NaN     NaN        NaN  1.000000e+08   

       ab_test vehicle_type  registration_year  gearbox       power_ps  model  \
count   100000        89901      100000.000000    94609  100000.000000  94508   
unique       2            8                NaN        2            NaN    248   
top       test    limousine                NaN  manuell            NaN   golf   
freq     51864        25827                NaN    74009            NaN   8048   
mean       NaN          NaN        2004.615250      NaN     114.600260    NaN   
std        NaN          NaN          94.080586      NaN     167.212918    NaN   
min        NaN          NaN        1000.000000      NaN       0.000000    NaN   
25%        NaN          NaN        1999.000000      NaN      69.000000    NaN   
50%        NaN          NaN        2003.000000      NaN     105.000000    NaN   
75%        NaN          NaN        2008.000000      NaN     150.000000    NaN   
max        NaN          NaN        9999.000000      NaN   17932.000000    NaN   

             odometer  registration_month fuel_type       brand  \
count   100000.000000       100000.000000     90946      100000   
unique            NaN                 NaN         7          40   
top               NaN                 NaN    benzin  volkswagen   
freq              NaN                 NaN     60132       21324   
mean    125432.800000            5.736830       NaN         NaN   
std      40203.126086            3.715986       NaN         NaN   
min       5000.000000            0.000000       NaN         NaN   
25%     100000.000000            3.000000       NaN         NaN   
50%     150000.000000            6.000000       NaN         NaN   
75%     150000.000000            9.000000       NaN         NaN   
max     150000.000000           12.000000       NaN         NaN   

       unrepaired_damage           ad_created  num_photos    postal_code  \
count              80644               100000    100000.0  100000.000000   
unique                 2                   87         NaN            NaN   
top                 nein  2016-04-03 00:00:00         NaN            NaN   
freq               70786                 3943         NaN            NaN   
mean                 NaN                  NaN         0.0   50827.250050   
std                  NaN                  NaN         0.0   25833.394784   
min                  NaN                  NaN         0.0    1067.000000   
25%                  NaN                  NaN         0.0   30457.000000   
50%                  NaN                  NaN         0.0   49549.000000   
75%                  NaN                  NaN         0.0   71673.750000   
max                  NaN                  NaN         0.0   99998.000000   

                  last_seen  
count                100000  
unique                68019  
top     2016-04-07 14:56:13  
freq                     11  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                     NaN

- Some of the columns that include text have nearly all the same values
  - seller
  - offer_type
  - num_photos

In [6]:
autos["num_photos"].value_counts()

0    100000
Name: num_photos, dtype: int64

In [7]:
autos.drop(columns=['seller','offer_type', 'num_photos', 'unrepaired_damage'])

date_crawled                                               name  \
0      2016-03-24 11:52:17                                         Golf_3_1.6   
1      2016-03-24 10:58:45                               A5_Sportback_2.7_Tdi   
2      2016-03-14 12:52:21                     Jeep_Grand_Cherokee_"Overland"   
3      2016-03-17 16:54:04                                 GOLF_4_1_4__3TÜRER   
4      2016-03-31 17:25:20                     Skoda_Fabia_1.4_TDI_PD_Classic   
5      2016-04-04 17:36:23  BMW_316i___e36_Limousine___Bastlerfahrzeug__Ex...   
6      2016-04-01 20:48:51                        Peugeot_206_CC_110_Platinum   
7      2016-03-21 18:54:38                       VW_Derby_Bj_80__Scheunenfund   
8      2016-04-04 23:42:13               Ford_C___Max_Titanium_1_0_L_EcoBoost   
9      2016-03-17 10:53:50  VW_Golf_4_5_tuerig_zu_verkaufen_mit_Anhaengerk...   
10     2016-03-26 19:54:18                                  Mazda_3_1.6_Sport   
11     2016-04-07 10:06:22      Volkswagen_Passat_Variant_2.0_TDI_Comfortline   
12     2016-03-15 22:49:09                  VW_Passat_Facelift_35i__"7Sitzer"   
13     2016-03-21 21:37:40                     VW_PASSAT_1.9_TDI_131_PS_LEDER   
14     2016-03-21 12:57:01  Nissan_Navara_2.5DPF_SE4x4_Klima_Sitzheizg_Blu...   
15     2016-03-11 21:39:15                       KA_Lufthansa_Edition_450€_VB   
16     2016-04-01 12:46:46                                        Polo_6n_1_4   
17     2016-03-20 10:25:19                        Renault_Twingo_1.2_16V_Aut.   
18     2016-03-23 15:48:05                   Ford_C_MAX_2.0_TDCi_DPF_Titanium   
19     2016-04-01 22:55:47                  Mercedes_Benz_A_160_Classic_Klima   
20     2016-04-01 19:56:48                  Volkswagen_Scirocco_1.4_TSI_Sport   
21     2016-03-27 11:38:00  BMW_530i_TÜV_7/17_Scheckheftgepflegt_sehr_gute...   
22     2016-03-23 14:52:51                      Opel_Meriva_1.Hand_TÜV_3.2018   
23     2016-03-12 19:43:07                                       Stadtflitzer   
24     2016-03-13 20:40:49                         MERCEDES_200E__TÜV_04/2016   
25     2016-03-18 21:44:09              BMW_530d_touring_Vollausstattung_NAVI   
26     2016-03-10 19:38:18                          Citroen_C4_Grand_Picasso.   
27     2016-03-07 12:51:23                     Honda_Civic_1.4_i_VTEC_Comfort   
28     2016-03-09 11:56:38                               Volkswagen_T3_andere   
29     2016-03-08 19:55:19                                     Fiat_Punto_1.2   
...                    ...                                                ...   
99970  2016-04-05 09:38:00               Fiat_Punto_1.2_8V_Active_Klimaanlage   
99971  2016-03-28 15:49:33                                 Kia_Picanto_1.1_EX   
99972  2016-03-22 21:52:51                         Opel_Corsa_1.2_TÜV_10/2017   
99973  2016-03-20 16:37:23                    Insignia_2.0_DIESEL_FULL_OPTION   
99974  2016-03-08 15:56:32                                      Audi_80_2.0_E   
99975  2016-03-25 09:54:05                     Citroën_Jumper_31_MH_YH5T_Club   
99976  2016-03-23 13:42:51            BMW_X3_3.0d_Volllerdr_Xenon_Frontschden   
99977  2016-03-30 01:58:06  Ford_C_Max_Ghia_Vollausstattung_Scheckheft_18_...   
99978  2016-03-19 10:47:05                           Renault_Laguna_Grandtour   
99979  2016-03-07 04:37:02                            Volkswagen_Scirocco_GT2   
99980  2016-03-31 10:57:03                       Opel_Zafira_2.2_DTI_Elegance   
99981  2016-03-08 18:40:39                         Ford_ECOSPORT_1.0_EcoBoost   
99982  2016-03-27 21:46:39                                 MINI_Cooper_Cabrio   
99983  2016-03-22 12:38:41                           Volkswagen_Caddy_2.0_SDI   
99984  2016-03-19 14:51:57                                Mercedes_C180_Sport   
99985  2016-03-06 16:25:24                                         BMW_320_Ci   
99986  2016-03-08 17:37:03            Corsa_1.7_CDTI_COSMO_5_TueREN_SUPER_TOP   
99987  2016-03-09 13:46:56                   

In [8]:
autos["price"].replace({"$",""})
autos["price"].replace({",",""})
autos["price"].astype(int)
                          
autos["price"].head()

0      480
1    18300
2     9800
3     1500
4     3600
Name: price, dtype: int64

In [9]:
autos.rename({"odometer": "odometer_km"}, axis=1, inplace=True)
autos["odometer_km"].head()
autos.columns 

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price', 'ab_test',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer_km', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'num_photos', 'postal_code',
       'last_seen'],
      dtype='object')

In [10]:
autos["odometer_km"].replace({"km",""})
autos["odometer_km"].replace({",",""})
autos["odometer_km"].astype(int)
                          
autos["odometer_km"].head()

0    150000
1    125000
2    125000
3    150000
4     90000
Name: odometer_km, dtype: int64

# Continuing to look into the odometer_km and price columns
## Things we will look for:
  - Max & min of the columns
  - look at the uniques values
  - remove any outliers using df[df["col"].between(x,y)]

In [11]:
autos["odometer_km"].value_counts()

150000    64564
125000    10311
100000     4325
90000      3295
80000      3032
70000      2674
60000      2369
50000      2084
5000       1863
40000      1730
30000      1664
20000      1535
10000       554
Name: odometer_km, dtype: int64

In [15]:
print(autos["price"].unique().shape) 
print(autos["price"].describe())
autos["price"].value_counts().head(20).sort_index(ascending = True)

(3231,)
count    1.000000e+05
mean     1.009095e+04
std      5.634880e+05
min      0.000000e+00
25%      1.150000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64


0       2917
500     1519
600     1009
650      805
700      797
750      873
800     1053
850      779
900      764
950      778
999      939
1000    1240
1100     773
1200    1235
1500    1464
1800     788
2000     981
2500    1167
3500     997
4500     785
Name: price, dtype: int64

In [18]:
autos["price"].value_counts().sort_index(ascending = False).head(20)


99999999    3
27322222    1
12345678    2
11111111    3
10000000    1
9999999     2
2995000     1
2795000     1
1250000     2
1111111     1
1000000     3
999999      2
911911      1
820000      1
600000      1
579000      1
500000      1
420000      1
395000      1
350000      1
Name: price, dtype: int64

In [19]:
autos["price"].value_counts().sort_index(ascending = True).head(20)

0     2917
1      308
2        3
3        2
4        1
5        8
7        1
8        4
9        1
10      21
11       1
12       3
13       2
14       2
15       6
16       1
17       3
20      14
24       1
25       8
Name: price, dtype: int64

In [20]:
#This will keep items that are within $1 and $200,000. 
#Leaving out the 2,917 items that have a price listed as free.
autos = autos[autos["price"].between(1,200000)]

In [22]:
autos["price"].describe()

count     97042.000000
mean       5894.340038
std        8429.487913
min           1.000000
25%        1250.000000
50%        3000.000000
75%        7499.000000
max      199900.000000
Name: price, dtype: float64

# Exploring the Date Columns
Below is a list of columns that hold dates.
- date_crawled
- registration_month
- registration_year
- ad_created
- last_seen

In [26]:
# data_crawled, last_seen and ad_created columns are all strings that need converted to float.
# let's explore the columns
autos[['date_crawled','ad_created','last_seen']].head()

date_crawled           ad_created            last_seen
0  2016-03-24 11:52:17  2016-03-24 00:00:00  2016-04-07 03:16:57
1  2016-03-24 10:58:45  2016-03-24 00:00:00  2016-04-07 01:46:50
2  2016-03-14 12:52:21  2016-03-14 00:00:00  2016-04-05 12:47:46
3  2016-03-17 16:54:04  2016-03-17 00:00:00  2016-03-17 17:40:17
4  2016-03-31 17:25:20  2016-03-31 00:00:00  2016-04-06 10:17:21

In [33]:
(autos["date_crawled"].str[:10]
 .value_counts(normalize=True, dropna=False)
 .sort_index().sort_values())

2016-04-07    0.001608
2016-04-06    0.003019
2016-03-18    0.013262
2016-04-05    0.013273
2016-03-06    0.014581
2016-03-13    0.015179
2016-03-05    0.025010
2016-03-24    0.029410
2016-03-27    0.029729
2016-03-16    0.030224
2016-03-17    0.031502
2016-03-26    0.031873
2016-03-22    0.032069
2016-03-31    0.032130
2016-03-23    0.032275
2016-03-11    0.032347
2016-03-25    0.032481
2016-03-10    0.032769
2016-03-08    0.033254
2016-03-30    0.033367
2016-03-15    0.033748
2016-04-01    0.034573
2016-03-09    0.034717
2016-03-29    0.034820
2016-04-02    0.034882
2016-03-07    0.034985
2016-03-28    0.035242
2016-03-19    0.035799
2016-03-14    0.035861
2016-03-21    0.035902
2016-03-12    0.036036
2016-03-20    0.036829
2016-04-04    0.037891
2016-04-03    0.039354
Name: date_crawled, dtype: float64

In [34]:
(autos["ad_created"].str[:10]
 .value_counts(normalize=True, dropna=False)
 .sort_index().sort_values())

2014-03-10    0.000010
2016-02-25    0.000010
2016-02-06    0.000010
2016-02-05    0.000010
2016-01-24    0.000010
2016-01-23    0.000010
2016-01-20    0.000010
2016-01-19    0.000010
2016-01-14    0.000010
2016-01-02    0.000010
2015-12-27    0.000010
2016-01-29    0.000010
2015-12-05    0.000010
2015-03-20    0.000010
2015-11-24    0.000010
2015-11-17    0.000010
2015-11-12    0.000010
2015-11-02    0.000010
2015-09-04    0.000010
2015-12-06    0.000010
2015-06-11    0.000010
2015-08-07    0.000010
2016-02-17    0.000021
2016-02-16    0.000021
2016-02-07    0.000021
2016-01-30    0.000021
2016-02-22    0.000021
2016-01-27    0.000021
2016-01-28    0.000021
2016-02-03    0.000021
                ...   
2016-03-06    0.015529
2016-03-13    0.016395
2016-03-05    0.022310
2016-03-27    0.029400
2016-03-24    0.029688
2016-03-16    0.030430
2016-03-17    0.031007
2016-03-22    0.031708
2016-03-26    0.032110
2016-03-23    0.032151
2016-03-31    0.032295
2016-03-11    0.032357
2016-03-25 

In [35]:
(autos["last_seen"].str[:10]
 .value_counts(normalize=True, dropna=False)
 .sort_index().sort_values())

2016-03-05    0.001164
2016-03-06    0.003844
2016-03-07    0.005091
2016-03-18    0.006997
2016-03-08    0.007821
2016-03-13    0.008244
2016-03-09    0.009583
2016-03-10    0.011335
2016-03-14    0.011974
2016-03-11    0.013334
2016-03-19    0.015756
2016-03-26    0.016148
2016-03-15    0.016302
2016-03-16    0.016477
2016-03-27    0.016776
2016-03-23    0.018209
2016-03-25    0.018837
2016-03-24    0.019198
2016-03-21    0.019662
2016-03-22    0.019971
2016-03-20    0.020239
2016-03-28    0.021918
2016-03-12    0.022866
2016-03-29    0.023114
2016-03-30    0.023279
2016-04-01    0.023598
2016-03-31    0.024165
2016-04-03    0.025020
2016-04-02    0.025298
2016-04-04    0.026020
2016-03-17    0.028761
2016-04-05    0.127574
2016-04-07    0.132067
2016-04-06    0.219359
Name: last_seen, dtype: float64

In [36]:
autos["registration_year"].describe()

count    97042.000000
mean      2004.620752
std         88.157584
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

# Registration year issues
 - The lowest registration year is 1000.
 - The largest registration year is 9999.
 
 - Find percentage of dates that fall outside of acceptable dates.
 - Exclude the dates outside of the acceptable limits.

In [47]:
# This will get the percentage of dates outside of the dates that are possible for 
((~autos["registration_year"].between(1900,2016)).sum() / autos.shape[0]) * 100

4.029653634012896

In [48]:
autos = autos[autos["registration_year"].between(1900,2016)]
autos["registration_year"].value_counts(normalize=True).head(20)

2000    0.071927
2005    0.068343
1999    0.068123
2006    0.062780
2001    0.062017
2003    0.060791
2004    0.059207
2002    0.058467
2007    0.054165
1998    0.053726
2008    0.050407
2009    0.048117
1997    0.043792
2010    0.038357
2011    0.037305
1996    0.032344
2012    0.029153
1995    0.028655
2016    0.028366
2013    0.019809
Name: registration_year, dtype: float64

It appears a majority of the cars were registered in the past 20 years.

In [51]:
autos["brand"].value_counts(normalize=True)

volkswagen        0.205132
bmw               0.111037
opel              0.107880
mercedes_benz     0.092974
audi              0.089158
ford              0.071153
renault           0.050500
peugeot           0.031870
fiat              0.026181
seat              0.020549
skoda             0.017103
mazda             0.016224
smart             0.015380
citroen           0.014698
nissan            0.013946
toyota            0.012790
hyundai           0.010731
mini              0.009980
volvo             0.008407
mitsubishi        0.008303
kia               0.007794
honda             0.006765
suzuki            0.006499
sonstige_autos    0.006302
alfa_romeo        0.006164
porsche           0.005354
chevrolet         0.004429
chrysler          0.003828
dacia             0.002729
daihatsu          0.002532
land_rover        0.002243
subaru            0.001989
jeep              0.001966
jaguar            0.001596
daewoo            0.001492
lancia            0.001341
saab              0.001260
r

It looks like the top 4 brands are all German brands. This makes me want to look into the original dataset and see where it originated. If you refer to the beginning where I changes the column names to snake_case. You can see that some of the information looks european (ex. odometer is in KM).

VW is the second largest car manufacturer in the world. The thing that doesn't make sense to me is how low Toyota appears on the list. 

Toyota is the largest car manufacturer in the work so you would expect them to be higher in the list. However, it is extremely important to remember that we are looking at Ebay Sales Data and since Toyota are known to be very reliable it might be less likely that people want to sell their Toyotas on Ebay. 

Toyota, being the largest manufacturer, might also offer better trade-in values for used cars at their dealerships due to their known reliablity. 

In [53]:
brand_counts = autos["brand"].value_counts(normalize = True)
common_brands = brand_counts[brand_counts > .01].index
print(common_brands)

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford', 'renault',
       'peugeot', 'fiat', 'seat', 'skoda', 'mazda', 'smart', 'citroen',
       'nissan', 'toyota', 'hyundai'],
      dtype='object')


In [58]:
brand_mean_price = {}
for brand in common_brands:
    car_brand = autos[autos["brand"] == brand]
    mean_price = car_brand["price"].mean()
    brand_mean_price[brand] = int(mean_price)
    
brand_mean_price

{'volkswagen': 5653,
 'bmw': 8745,
 'opel': 3004,
 'mercedes_benz': 8895,
 'audi': 9650,
 'ford': 3622,
 'renault': 2456,
 'peugeot': 3276,
 'fiat': 2818,
 'seat': 4784,
 'skoda': 6631,
 'mazda': 4346,
 'smart': 3674,
 'citroen': 3648,
 'nissan': 4559,
 'toyota': 5222,
 'hyundai': 5666}

In [ ]:
bmp_series = pd.Series(brand_mean_prices)
pd.DataFrame(bmp_series, columns=["mean_price"]):
bmp_series = pd.Series(brand_mean_prices)
pd.DataFrame(bmp_series, columns=["mean_price"])

Out of the top brands BMW, Mercedes-Benz, and Audi prices reflect that they are luxury brands. Which surprises me about Skoda, because it is almost the price of BMW but I have never heard of it.

In [63]:
brand_mean_mileage = {}

for brand in common_brands:
    car_brand = autos[autos["brand"] == brand]
    mean_mileage = car_brand["odometer_km"].mean()
    brand_mean_mileage[brand] = int(mean_mileage)
    
mean_mileage = pd.Series(brand_mean_mileage).sort_values(ascending=False)
mean_prices = pd.Series(brand_mean_price).sort_values(ascending=False)

In [64]:
brand_info = pd.DataFrame(mean_mileage,columns=['mean_mileage'])
brand_info

mean_mileage
bmw                  132279
mercedes_benz        129361
opel                 129206
audi                 128728
volkswagen           128143
renault              127860
peugeot              124473
ford                 124224
mazda                123496
citroen              121219
nissan               119937
seat                 119760
toyota               117549
fiat                 117305
skoda                112160
hyundai              104251
smart                 99560

In [65]:
brand_info["mean_price"] = mean_prices
brand_info

mean_mileage  mean_price
bmw                  132279        8745
mercedes_benz        129361        8895
opel                 129206        3004
audi                 128728        9650
volkswagen           128143        5653
renault              127860        2456
peugeot              124473        3276
ford                 124224        3622
mazda                123496        4346
citroen              121219        3648
nissan               119937        4559
seat                 119760        4784
toyota               117549        5222
fiat                 117305        2818
skoda                112160        6631
hyundai              104251        5666
smart                 99560        3674

# Next Steps
## Data cleaning & Analysis next steps:
 - Identify categorical data that uses german words, translate them and map the values to their english counterparts
 - Convert the dates to be uniform numeric data, so "2016-03-21" becomes the integer 20160321.
 - See if there are particular keywords in the name column that you can extract as new columns
 - Find the most common brand/model combinations
 - Split the odometer_km into groups, and use aggregation to see if average prices follows any patterns based on the milage.
 - How much cheaper are cars with damage than their non-damaged counterparts?